test/index.js

```diff
@@ -173,27 +217,32 @@ test('time out when there is no response', function (t) {
 
 test('decoding something that is gzipped', function (t) {
   var json = JSON.stringify(JSON.parse(fs.readFileSync('./test/fixtures/uncompressed.json')))
-  var gzipped = zlib.gzipSync(json)
-  var data = [gzipped]
-  var res = {headers: {'content-encoding': 'gzip'}}
-  var service = new FeatureService('http://service.com/mapserver/2')
-
-  service._decode(res, data, function (err, json) {
-    t.equal(json.features.length, 2000)
-    t.end()
+  zlib.gzip(json, function (err, gzipped) {
+    if (err) t.end()
+    var data = [gzipped]
+    var res = {headers: {'content-encoding': 'gzip'}}
+    var service = new FeatureService('http://service.com/mapserver/2')
+
+    service._decode(res, data, function (error, json) {
+      t.equal(error, null)
+      t.equal(json.features.length, 2000)
+      t.end()
+    })
   })
 })
 
 test('decoding something that is deflated', function (t) {
   var json = JSON.stringify(JSON.parse((fs.readFileSync('./test/fixtures/uncompressed.json'))))
-  var deflated = zlib.deflateSync(json)
-  var data = [deflated]
-  var res = {headers: {'content-encoding': 'deflate'}}
-  var service = new FeatureService('http://service.com/mapserver/2')
-
-  service._decode(res, data, function (err, json) {
-    t.equal(json.features.length, 2000)
-    t.end()
+  zlib.deflate(json, function (err, deflated) {
+    if (err) t.end()
```

Comment:
Ah, I think I need a t.plan(). If I do that, ending the test here will cause a fail, which is what I want.